Minesweeper.py

```diff
@@ -44,132 +70,143 @@ def __init__(self):
             "The numbers say how many mines are adjacent to the tile cleared",
             "",
             "Press R to restart",
+            "Press ESCAPE to quit"
             "",
             "Press SPACE to start the game.",
         ]
 
-    def draw(self): # draws the tutorial screen
-        # DHB: using win as a global variable is a bad practice. It is better to send
-        # DHB: things like this into constructor args and store in member variables.
-        # DHB: When you start unit testing, global variables are a class of hell
-        # DHB: also, pretty sure you should only fill the window with the background color in the top-level draw function
-        win.fill(Color.BACKGROUND)
-        for i, instruction in enumerate(self.instructions):
-            text = self.font.render(instruction, True, self.text_color)
-            text_rect = text.get_rect(center=(win.get_width() // 2, 100 + i * 30))
-            win.blit(text, text_rect)
-        # DHB: I'm pretty certain only the top-level draw function should call display.update 
-        pygame.display.update() # Update the display
+    def draw(self):
+        """ draws the tutorial screen using the text information from the init function
+        """
+        self.game.win.fill(Color.BACKGROUND)
+        for i, message in enumerate(self.message):
+            x = (self.game.win.get_width() // 2)
+            y = ( 100 + i * 30)
+            text = self.font.render(message, True, self.text_color)
+            text_rect = text.get_rect(center=(x, y))
+            self.game.win.blit(text, text_rect)
 
     def handle_events(self, events):
+        """ detects a SPACE press and causes the transition from the tutorial state to the game state
+        """
         for event in events:
             if event.type == pygame.KEYDOWN:
                 if event.key == pygame.K_SPACE:
-                    # Transition to the game state
-                    # DHB: game is a separate object. one should never modify the contents of member variables
-                    # DHB: directly on another object. Instead call a function such as "set_state".
-                    game.state = "game"
-
-class GameWonScreen(object): # creates the victory screen
-    def __init__(self):
-        self.font = pygame.font.Font(None, 30) # Font for win text
-        self.text_color = Color.BLACK # Color for win text
-        # DHB: instructions is not a reasonable variable name here
-        self.instructions = [
+                    self.game.gamestate.set_state_game()
+                    self.game.gamestate.set_state_tutorial()
```

Comment:
rather than toggling these two variables, when you set the state to be "game = True", the function should just set all other states to be False.